### DISCUSSION TITLE
$PATH errors, no matter how I edit my ~/.zshrc and .zprofile "zsh: no such file or directory: /opt/homebrew/opt/curl/bin:/opt/homebrew/bin:/opt/homebrew/bin:/opt/homebrew/sbin:/usr/local/bin:/System/Cryptexes/App/usr/bin:/usr/bin:/bin:/usr/sbin:/sbin:/Library/Apple/usr/bin"

### DISCUSSION THREAD

**Discussion by dmoney96** [Homebrew/discussions#4285] [Everyday usage]
### Output of `brew config`

```shell
HOMEBREW_VERSION: 4.0.4
ORIGIN: https://github.com/Homebrew/brew
HEAD: 4d0154fa7926e8a7b06e9b074dd45957e820407c
Last commit: 32 hours ago
Core tap JSON: 28 Feb 21:13 UTC
HOMEBREW_PREFIX: /opt/homebrew
HOMEBREW_CASK_OPTS: []
HOMEBREW_MAKE_JOBS: 12
Homebrew Ruby: 2.6.10 => /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/bin/ruby
CPU: dodeca-core 64-bit arm_blizzard_avalanche
Clang: 14.0.3 build 1403
Git: 2.39.2 => /Library/Developer/CommandLineTools/usr/bin/git
Curl: 7.86.0 => /usr/bin/curl
macOS: 13.2.1-arm64
CLT: 14.3.0.0.1.1676486335
Xcode: 14.2 => /Users/dominiquemiller/Downloads/Xcode.app/Contents/Developer
Rosetta 2: false
```

### Output of `brew doctor`

```shell
Your system is ready to brew.
```

### Description of issue

No matter what I do, every time I check my $PATH on my Macbook Pro M2, I get the following message. I've tried nano, vim, text editor to edit ~/.zshrc and .profile and .zprofile and to no avail. I am totally out of ideas, and I need to fix this issue. I've set different path variables, based on suggestions here and on reddit, and based on the ohmyzsh repo, and to absolutely no success. Here is what my zshrc and profile files look like currently. 
[zshrcforgithub.txt](https://github.com/Homebrew/discussions/files/10854990/zshrcforgithub.txt)
[zprofileforgithub.txt](https://github.com/Homebrew/discussions/files/10855049/zprofileforgithub.txt)

**Comment by dmoney96** [comment]
I've made some changes, and this is the fun new error message I get, upon login to my shell $/Users/$USER/.zprofile:52: not a directory: /opt/homebrew/bin/brew/ruby

**Reply by gromgit** [reply]
The `.zprofile` you posted seem to have been saved with formatting in RTF (TextEdit?). Cutting out all the unwanted formatting reveals this weird construct therein:
```
path=(
  $HOME/{,s}bin
  eval $(/opt/homebrew/bin/brew/ruby shellenv)
  $path
)
```
There are at least two problems with the above:
1. There's no such executable as `/opt/homebrew/bin/brew/ruby`. The instructions given by the installer were to add `eval "$(/opt/homebrew/bin/brew shellenv)"` to your `.zprofile`. Note the correct executable pathname and the double quotes.
2. The above doesn't do what you think it does. Try adding a `declare -p path` immediately after the above to see exactly what your `path` array ends up being set to.

The cleanest solution to your current issue: _delete_ the `shellenv` line from the above, and _replace_ this line near the bottom of your `.zprofile`:
```
export PATH="/opt/homebrew/bin:$PATH"
```
with _exactly_ this command:
```
eval "$(/opt/homebrew/bin/brew shellenv)"
```
